### DISCUSSION TITLE
xDrip only updates once every 10 minutes - is there a way to increase this frequency? [dexcom g7 + android]

### DISCUSSION THREAD

**Discussion by bkuz114** [NightscoutFoundation/xDrip#3747] [General]
So sorry if this has been asked before, I searched discussions, but couldn't find anything. Deeply grateful for xDrip - it's an excellent app.

Currently with my dexcom g7, xDrip only updates BG readings once every 10 minutes. Is there a way to increase this frequency? I saw a [reddit post](https://old.reddit.com/r/Freestylelibre/comments/1bdyv86/xdrip_libre_3_web_follower_question/kuwf695/) saying they were able to increase the frequency on their iphone, but I can't find such a setting (settings --> general) on my Android. Is this something that's only possible on iphone?

In case helpful, it appears my xDrip version is 0bd5d9b-2024.08.02 (I found this on hamburger menu --> system status --> classic status page)

**Comment by Navid200** [comment]
Please post a screenshot of the Dex status page to start.

**Reply by bkuz114** [reply]
![Screenshot_20241101_192716_xDrip+](https://github.com/user-attachments/assets/885bcd2b-99f3-443c-b67f-3e9807c055a3)

**Reply by bkuz114** [reply]
Note that it's currently disconnected, hence why it's saying last connected 13 minutes ago. (Not sure why it disconnected, it has issues when I'm using my bluetooth headphones and they disconnect. It will connect again eventually.)

**Reply by Navid200** [reply]
What is the phone brand and model?

**Reply by bkuz114** [reply]
Brand: Samsung Galaxy A14 5G

Model: SM-S146VL

**Reply by Navid200** [reply]
Post a screenshot of the following page:
`Settings` > `Dex Debug Settings`

**Reply by bkuz114** [reply]
![Screenshot_20241101_193545_xDrip+](https://github.com/user-attachments/assets/1a2ce2a8-945a-49be-8bb1-a5a0c68dedac)
![Screenshot_20241101_193550_xDrip+](https://github.com/user-attachments/assets/8b93eec4-d69b-4435-b660-29ed4006b907)

**Reply by Navid200** [reply]
Enable `Special Pairing Workaround`

**Reply by bkuz114** [reply]
(Note that I set that up according to instructions linked to on [this page](https://navid200.github.io/xDrip/docs/Dexcom/G7.html))

**Reply by bkuz114** [reply]
Interesting about `Special Pairing Workaround`. I have now enabled it. What's the expected result of doing that, and do I need to restart the app to get the result?

(Note that I had left it unchecked as per the screenshot instructions [here](https://navid200.github.io/xDrip/docs/images/g6-recommended-settings.png))

**Reply by Navid200** [reply]
That is included in those instructions.  I am happy to edit it if it is not.

**Reply by Navid200** [reply]
What you are experiencing is intermittent connectivity:
https://navid200.github.io/xDrip/docs/Intermittent.html

I admit that the guide is not perfect and even not clear at times.  It is not easy to create a guide that is easy to follow that also addresses all possible conditions and is short.
So, I have ended up with several pages.  But, how do I make sure that everyone sees every page ...

I will work on it to improve it.

**Reply by Navid200** [reply]
If it still has not reconnected, it may help to restart the phone.

**Reply by Navid200** [reply]
Samsung is a phone manufacturer that modifies Android too much.
We used to have a setting is xDrip called `Samsung Workarounds`.
And now, we have this setting with the summary reading `Some Samsung devices ...`.  

It is not possible to have an image showing the settings for all phones because all phones are not Samsung and Samsung is not all phones.
If it was to me, I would probably state that xDrip does not support Samsung.  But, fortunately for everyone, it is not up to me.
It is not easy to create a simple short guide for every single phone.

I am open to suggestions.

**Reply by bkuz114** [reply]
Oh yes, the intermittent connectivity. Thank you. When I experience that, usually it reconnects on its own after 10 minutes or so, but sometimes I just restart the collector and restart the phone, and then things work again. Hopefully this setting will help things - so thanks!

By the way - you probably did mention that in the guide, I might have just skipped it.

I think your guide is good - I was able to get dexcom g7 running because of that guide. I imagine it's not a trivial task to develop a guide that works for all phones. Also, I can tell you are trying to re-use instructions (instead of duplicating things), which makes sense, because if something changes you don't want to be updating all these various instructions pages and missing one. But of course this then makes things a little difficult because you're relying on lots of links.

By the way - I suspect this particular model of Samsung has bluetooth issues in general. For example, just with my bluetooth headphones, they're always disconnecting randomly, interestingly if I switch between apps it happens. I think there's just complications with this phone.

**Reply by Navid200** [reply]
With special pairing workaround enabled, you should get one reading every 5 minutes consistently.
If it disconnects, it will not be once every 10 minutes any more.  It may be once every other hour.  And that, you should hopefully be able to address by figuring out how to completely disable battery optimization.

**Reply by bkuz114** [reply]
>  If it was to me, I would probably state that xDrip does not support Samsung. But, fortunately for everyone, it is not up to me.

Oh, if there's any advice I can offer, it's to please not say this, because it *does* work with some Samsung phones (it works with mine for example). I think a better thing to say might be: "it might or might not work, because there's complications with Samsung, but it's worth a try and here's the best way to try it." If I'd seen something saying "Samsung not supported", I wouldn't have even tried, and then I wouldn't be using xDrip, which my life pretty much depends on at this point.

So far xDrip works fantastically for this g7. Yes, it disconnects sometimes, but not often. Barring me physically going out of bluetooth range, I might get a disconnect every couple days and then it reconnects again. Not a big deal.

**Reply by bkuz114** [reply]
>With special pairing workaround enabled, you should get one reading every 5 minutes consistently.

Fantastic news! Just checked, and yes, I'm getting readings every 5 minutes now! Excellent!

> If it disconnects, it will not be once every 10 minutes any more. It may be once every other hour. And that, you should hopefully be able to address by figuring out how to completely disable battery optimization.

I think I mis-communicated. It does not disconnect every 10 minutes - maybe I get a disconnect once every couple days. It was just coincidence that when I posted that screenshot, it had disconnected. Really, it only happens rarely. Batter optimization is disabled both for xDrip and bluetooth. I think this phone just have bluetooth issues in general.

**Reply by Navid200** [reply]
Of course, there was a little bit of humor in that.
Samsung is popular.  That is why our developer has spent so much time addressing issues caused only by Samsung.

**Reply by bkuz114** [reply]
Ah I understand now. Sorry for missing the sarcasm, heh it's difficult to pick up sometimes on github

Yes, Samsung is affordable, from what I can see. I had to get a phone specifically for using CGM (the libre 3 actually), as my actual phone doesn't have NFC. When I started looking for a phone to purchase, I was shocked to see how expensive many are (I saw phones that were $800 USD). This was pretty much the only phone i could find for less than $100 USD which had NFC and would work with the CGM. If I had to get an iphone or something to use the CGM, it wouldn't have been possible. So I really appreciate that the developer puts effort into getting things to work with samsung.

I have a reader for the dexcom g7 so not a big deal anymore. But if I want to use those free style libres I need this phone.

**Reply by bkuz114** [reply]
As an update, interestingly, since enabling that option (`Special Pairing Workaround`), I'm actually experiencing a lot of disconnects. I've had 2 in the last couple hours, and it takes 20-30 minutes for it to reconnect. Is there any thoughts why this might be? (Or maybe it's just coincidence?)

**Reply by Navid200** [reply]
Do you have trust auto-connect disabled or enabled?

**Reply by bkuz114** [reply]
disabled, as per instructions [here](https://navid200.github.io/xDrip/docs/images/ble-recommended-stngs.png)

**Reply by Navid200** [reply]
As per the guide advice, try enabling it: https://navid200.github.io/xDrip/docs/Intermittent.html

**Reply by bkuz114** [reply]
Thanks. I hadn't seen that link yet. I enabled it, and so far no further disconnections!
